drakrun/drakrun/lib/postprocessing/capa_plugin/capa_processor.py

```diff
@@ -0,0 +1,428 @@
+import itertools
+import logging
+import multiprocessing
+import shutil
+import zipfile
+from pathlib import Path
+from typing import Any, BinaryIO, Dict, Iterator, List, Optional, Tuple, Union
+
+import capa
+import capa.capabilities
+import capa.capabilities.common
+import capa.engine
+import capa.features
+import capa.features.address
+import capa.features.common
+import capa.features.extractors
+import capa.features.extractors.base_extractor
+import capa.features.extractors.drakvuf
+import capa.features.extractors.drakvuf.extractor
+import capa.helpers
+import capa.loader
+import capa.main
+import capa.render
+import capa.render.json
+import capa.render.result_document
+import capa.rules
+import orjson
+
+# rules related configuration
+capa_rules_dir = Path("./capa-rules")
+
+# analysis configuration
+analyze_malware_pids_only = False
+perform_static_analysis = False
+perform_dynamic_analysis = True
+
+logger = logging.getLogger(__name__)
+
+
+def check_rules_directory_exist(path: Path) -> bool:
+    # this method checks whether a non-empty directory exists
+    return path.is_dir() and any(path.iterdir())
+
+
+def find_process_in_pstree(pstree: List, pid: int, procname: str) -> Dict:
+    # this methods searches for a process in the generated process_tree.json file
+    for process in pstree:
+        if process["pid"] == pid:
+            return process
+        else:
+            child_processes = find_process_in_pstree(process["children"], pid, procname)
+            if child_processes:
+                return child_processes
+
+    # return None in case the process is not found
+    return None
+
+
+def get_all_child_processes(process: Dict) -> Iterator[int]:
+    # this method returns all of the child processes pids (recursively), including their parent's
+    for p in process["children"]:
+        yield from get_all_child_processes(p)
+
+    # yield current node's pid
+    yield process["pid"]
+
+
+def get_rules(rules_dir: List[Path]) -> Optional[capa.rules.RuleSet]:
+    try:
+        rules = capa.rules.get_rules(rules_dir)
+    except (IOError, capa.rules.InvalidRule, capa.rules.InvalidRuleSet):
+        # display the official capa error message in case there exists any malformed capa rules
+        rules = None
+        logger.exception(
+            "Make sure your file directory contains properly formatted capa rules. You can download the standard "
+            + "collection of capa rules from https://github.com/mandiant/capa-rules/releases."
+        )
+        logger.error(
+            "Please ensure you're using the rules that correspond to your major version of capa (%s)",
+            capa.version.get_major_version(),
+        )
+        logger.error(
+            "Or, for more details, see the rule set documentation here: %s",
+            "https://github.com/mandiant/capa/blob/master/doc/rules.md",
+        )
+
+    return rules
+
+
+def filter_rules(
+    rules: capa.rules.RuleSet, filter_function=lambda rule: rule
```

Comment:
`filter: None` works the same way, and in general it's a bad practice to use complex default values:
```
>>> list(filter(None, [1, 2, 3]))
[1, 2, 3]
```
so maybe:
```suggestion
    rules: capa.rules.RuleSet, filter_function=None
```